src/github.com/matrix-org/dendrite/clientapi/routing/register.go

```diff
@@ -270,10 +272,12 @@ func LegacyRegister(
 			return util.MessageResponse(403, "HMAC incorrect")
 		}
 
-		return completeRegistration(req.Context(), accountDB, deviceDB, r.Username, r.Password)
+		// TODO: does the legacy registration request support initial
+		// display name?
+		return completeRegistration(req.Context(), accountDB, deviceDB, r.Username, r.Password, "legacy registration")
 	case authtypes.LoginTypeDummy:
 		// there is nothing to do
-		return completeRegistration(req.Context(), accountDB, deviceDB, r.Username, r.Password)
+		return completeRegistration(req.Context(), accountDB, deviceDB, r.Username, r.Password, "legacy registration")
```

Comment:
Instead of having hard coded values I'd be tempted to allow nils that get inserted as nulls

Reply:
Great, I'll make the needed changes